lamagraph-core/src/Core/Node.hs

```diff
@@ -0,0 +1,95 @@
+module Core.Node where
+
+import Clash.Prelude
+import Data.Maybe (isJust)
+
+type Address = Unsigned 16
+
+-- | Pointer on port of the `Node` with visited flag.
+data Port = Port
+  { _targetAddress :: Address
+  , _edgeIsVisited :: Bit
+  }
+  deriving (NFDataX, Generic)
+
+-- | Node in the RAM.
+data Node numberOfPorts = Node
+  { _primaryPort :: Port
+  , _secondaryPorts :: Vec numberOfPorts (Maybe Port)
```

Comment:
Why do we need `Maybe` here?

Reply:
In our current realization -- yes, because there is fixed number of secondary ports in node. So this number should be the maximum of the number of ports of all nodes. Nothing means, that port is actually absent.

Reply:
I'm pretty sure that we can restrict `Vector` length using some type magic, but I'm fine with this way if it's easier to work with when prototyping